controllers/multiclusterobservability/multiclusterobservability_status.go

```diff
@@ -19,6 +20,87 @@ import (
 	"github.com/open-cluster-management/multicluster-observability-operator/pkg/config"
 )
 
+var (
+	stopStatusUpdate            = make(chan struct{})
+	stopCheckReady              = make(chan struct{})
+	requeueStatusUpdate         = make(chan struct{})
+	updateStatusIsRunnning      = false
+	updateReadyStatusIsRunnning = false
+)
+
+// Start goroutines to update MCO status
+func StartStatusUpdate(c client.Client, instance *mcov1beta2.MultiClusterObservability) {
+	if !updateStatusIsRunnning {
+		go func() {
+			updateStatusIsRunnning = true
+			// defer close(stopStatusUpdate)
+			// defer close(requeueStatusUpdate)
+			for {
+				select {
+				case <-stopStatusUpdate:
+					updateStatusIsRunnning = false
+					stopCheckReady <- struct{}{}
+					log.V(1).Info("status update goroutine is stopped.")
+					return
```

Comment:
nit: `return` is not needed.

Reply:
use `return` to break this endless `for`.